src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
```

Comment:
Unless we allocate memory in a shared memory area I don't see it possible to allocate on the heap and send the data to another process.  Of course I might be missing something obvious...

Let's deal with that afterwards though. Initial testing seems to be that this is just plain faster than zmq so the one copy might be OK.